NewsPortal.BusinessLogic/Services/UserService.cs

```diff
@@ -0,0 +1,192 @@
+﻿using NewsPortal.BusinessLogic.Constants;
+
+namespace NewsPortal.BusinessLogic.Services
+{
+    public class UserService : IUserService
+    {
+        private readonly IMapper? _mapper;
+        private readonly IResponseFactory _responseFactory;
+        private readonly IRoleService _roleService;
+        private readonly UserManager<User>? _userManager;
+
+        public UserService(
+            IMapper? mapper, 
+            IResponseFactory responseFactory, 
+            IRoleService roleService, 
+            UserManager<User>? userManager)
+        {
+            _mapper = mapper;
+            _responseFactory = responseFactory;
+            _roleService = roleService;
+            _userManager = userManager;
+        }
+
+        public async Task<BaseResponse> Register(RegisterModel registerModel)
+        {
+            var user = new User
+            {
+                Email = registerModel.Email,
+                UserName = registerModel.UserName,
+                CreatedDate = DateTime.UtcNow,
+                UpdatedDate = DateTime.UtcNow
```

Comment:
Don't need to set UpdatedDate in all create actions, use only in Update actions. Make UpdateDate properties in ITimeStamp nullable